the-blue-alliance-ios/Operations/App Setup/AppSetupOperation.swift

```diff
@@ -0,0 +1,76 @@
+import CoreData
+import Foundation
+import UIKit
+
+class AppSetupOperation: TBAOperation {
+
+    var window: UIWindow
+    var rootSplitViewController: UISplitViewController
+    var persistentContainerOperation: PersistentContainerOperation
+    var remoteConfigServiceOperation: RemoteConfigServiceOperation
+
+    let appSetupOperationQueue = OperationQueue()
+
+    init(window: UIWindow, rootSplitViewController: UISplitViewController, persistentContainer: NSPersistentContainer, remoteConfigService: RemoteConfigService) {
+        self.window = window
+        self.rootSplitViewController = rootSplitViewController
+        self.persistentContainerOperation = PersistentContainerOperation(persistentContainer: persistentContainer)
+        self.remoteConfigServiceOperation = RemoteConfigServiceOperation(remoteConfigService: remoteConfigService)
+
+        super.init()
+
+        self.persistentContainerOperation.completionBlock = { [unowned persistentContainerOperation] in
+            if let error = persistentContainerOperation.completionError as NSError? {
+                self.completionError = error
+            }
+        }
+    }
+
+    override func execute() {
+        let blockOperation = BlockOperation {
+            if self.completionError == nil {
+                self.propogatePersistentContainer(to: self.rootSplitViewController)
+
+                DispatchQueue.main.async {
```

Comment:
Let's move ALL of this back in to the App Delegate - it feels very strange poking around in UIKit stuff inside of here. It also makes it extremely difficult to test